components/script/fetch.rs

```diff
@@ -99,30 +99,29 @@ fn from_referrer_to_referrer_url(request: &NetTraitsRequest) -> Option<ServoUrl>
 }
 
 fn request_init_from_request(request: NetTraitsRequest) -> NetTraitsRequestInit {
-    NetTraitsRequestInit {
-        method: request.method.clone(),
-        url: request.url(),
-        headers: request.headers.clone(),
-        unsafe_request: request.unsafe_request,
-        body: request.body.clone(),
-        destination: request.destination,
-        synchronous: request.synchronous,
-        mode: request.mode.clone(),
-        use_cors_preflight: request.use_cors_preflight,
-        credentials_mode: request.credentials_mode,
-        use_url_credentials: request.use_url_credentials,
-        origin: GlobalScope::current()
-            .expect("No current global object")
-            .origin()
-            .immutable()
-            .clone(),
-        referrer_url: from_referrer_to_referrer_url(&request),
-        referrer_policy: request.referrer_policy,
-        pipeline_id: request.pipeline_id,
-        redirect_mode: request.redirect_mode,
-        cache_mode: request.cache_mode,
-        ..NetTraitsRequestInit::default()
-    }
+    NetTraitsRequestInit::new(request.url())
```

Comment:
I would actually like to revert the change to this function, and remove the use of `..NetTraitsRequesInit::default()`. That will make the compiler catch when we add members to NetTraitsRequestInit and need to be sure they are initialized correctly.

Reply:
Sorry, I didn't get you.

Reply:
I would like to revert this particular change: this method should construct a new NetTraitsRequestInit object with an explicit structvliteral that initializes every field (instead of using `..NetTraitsRequestInit::default()`).

Reply:
What do you think about a method overload to the method to the BuilderClass new passing all the arguments?

Reply:
What do you think now?

Reply:
Then we will be able to construct this object without setting the properties directly removing the "pub" from them.

e.g.:
```
RequestBuilder {
    url: url
    ...
}
```
It will throw an error.

Reply:
@jdm

Reply:
The advantage of allowing some code like `request_init_from_request` to create the RequestInit value directly is that it's more easy to understand code like `use_url_credentials: false` instead of a `false` value that is one of 20 arguments. I think it's fine to keep the field values of RequestInit public and build it directly, rather than adding a method.

Reply:
yeah, i've got you.